### PR TITLE
ENG-713 ENG-712 add new endpoints for configuration and context configuration

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
New endpoint to get general configuration:
```
GET http://localhost:8080/v1/configuration
{
  "workspace": {
     "network": {
        "plural_dns": true,
        "subdomain": "lukasz.onplural.sh"
    },
   "bucket_prefix": "lukasz",
   "cluster": "lukasz"
  },
  "git": {
     "url": "git@github.com:zreigz/kind-test.git",
     "root": "/home/lukasz/plural/kind-test",
     "name": "kind-test",
     "branch": "main"
   }
}

```

New endpoint to patch context configuration
```
POST http://localhost:8080/v1/context/configuration 

{
  "configuration": {
    "bootstrap": {},
    "console": {
      "admin_email": "lukasz@plural.sh",
      "admin_name": "lukasz",
      "console_dns": "console.lukasz.onplural.sh",
      "git_email": "zreigz@gmail.com",
      "git_user": "zreigz",
      "passphrase": "",
      "repo_url": "git@github.com:zreigz/kind-test.git"
    },
    "etcd": {},
    "ingress-nginx": {},
    "minio": {
      "consoleHostname": "minioui.lukasz.onplural.sh",
      "hostname": "minio.lukasz.onplural.sh"
    },
    "monitoring": {},
    "postgres": {
      "wal_bucket": "lukasz-lukasz-postgres-wal"
    }
  }
}

```

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [X] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/server/configuration.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"net/http"
+	"path/filepath"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pluralsh/plural/pkg/manifest"
+	"github.com/pluralsh/plural/pkg/utils/git"
+)
+
+type Configuration struct {
```
Comment (michaeljguarino) [inline pkg/server/configuration.go:12]: One thing i just remembered, we should also return the current context configuration here as well, probably just as a `map[string]interface{}`

**Review by michaeljguarino — COMMENTED** [review]
lgtm overall, just one small change as I realized we're going to need a bit more in the configuration endpoint

File: pkg/server/configuration.go (hunk 1 above)
Comment (zreigz) [reply pkg/server/configuration.go:12]: OK, no problem

**Comment by zreigz** [comment]
@michaeljguarino PTAL

File: pkg/server/configuration.go (hunk 1 above)
Comment (zreigz) [reply pkg/server/configuration.go:12]: done

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**